platform/mellanox/smart-switch/dpuctl/dpuctl_hwm.py

```diff
@@ -0,0 +1,195 @@
+"""Class Implementation for per DPU functionality"""
+import os.path
+
+try:
+    from .inotify_helper import InotifyHelper
+    from sonic_py_common.logger import Logger
+    from sonic_platform import utils
+except ImportError as e:
+    raise ImportError(str(e)) from e
+
+EVENT_BASE = "/var/run/hw-management/events/"
+SYSTEM_BASE = "/var/run/hw-management/system/"
+CONFIG_BASE = "/var/run/hw-management/config"
+logger = Logger()
+
+WAIT_FOR_SHTDN = 60
+WAIT_FOR_DPU_READY = 60
+
+
+class MyMessageWriter():
+    """Class for writing data to files"""
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.file_obj = None
+        if not os.path.isfile(self.file_name):
+            raise FileNotFoundError(f"File {self.file_name} does not exist!")
+
+    def __enter__(self):
+        self.file_obj = open(self.file_name, 'w', encoding="utf-8")
+        return self.file_obj
+
+    def __exit__(self, *args):
+        self.file_obj.close()
+
+
+def write_file(file_name, content_towrite, dpu_name):
```

Comment:
This can be converted into a class method

Reply:
Done